File: ms_relatorio/src/main/java/com/saude_mais/ms_relatorio/entities/Agendamento.java
package com.saude_mais.ms_relatorio.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor

public class Agendamento {

    private String id;
    private String idPac;
    private String dataConsulta;
    private String nomeMedico;
    private String especialidadeMedico;

}

File: validationSchema/ms_agendamento/agendamento.js
use clinica;

db.createCollection("agendamento",
    {
        validator: {
            $jsonSchema: {
                required: [
                    "dataConsulta",
                    "nomeMedico",
                    "especialidadeMedico",
                    "action"
                ], properties: {
                    dataConsulta: {
                        bsonType: "string", maxLength: 10, description: "Data da consulta."
                    }, nomeMedico: {
                        bsonType: "string", maxLength: 30, description: "Nome do medico.",
                        }, especialidadeMedico:{
                            bsonType: "string", maxLength: 30, description: "Especialidade.",
                            }, action:{
                                bsonType: ["object"], description: "Ação.",
                            }
                        }
                    }
                }
            });





;